src/analyzing/Constant_propagation.ml

```diff
@@ -679,29 +666,30 @@ let propagate_basic lang prog =
             let assigned_just_once =
               is_assigned_just_once stats.var_stats (H.str_of_ident id, sid)
             in
-            (if
-             H.has_keyword_attr Const attrs
-             || H.has_keyword_attr Final attrs
-             || (assigned_just_once && is_js env)
-             || assigned_just_once && is_lang env Lang.Java
-                && List.exists is_private attrs
-            then
-             match (!id_svalue, e.e) with
-             (* When the name already has an svalue computed, just use
-              * that. DeepSemgrep assigns svalues sometimes in its naming
-              * phase. *)
-             | Some svalue, _ -> add_constant_env id (sid, svalue) env
-             | None, L literal -> add_constant_env id (sid, Lit literal) env
-             (* For any other symbolic expression, it is OK to propagate it symbolically so long as
-                the lvalue is only assigned to once.
-                Although we may propagate expressions with identifiers in them, those identifiers
-                will simply not have an `svalue` if they are non-propagated as well.
-             *)
-             | None, _
-               when Dataflow_svalue.is_symbolic_expr e
-                    && no_cycles_in_sym_prop sid e ->
-                 add_constant_env id (sid, Sym e) env
-             | None, _ -> ());
+            if
```

Comment:
The core change here is that we add `id_flags := IdFlags.set_final !id_flags;`, the rest is code reformatting.